framework/test/unit/modules/dynamic_rewards/module.spec.ts

```diff
@@ -0,0 +1,384 @@
+					},
+					generatorConfig: {},
+				});
+			} catch (error: any) {
+				// eslint-disable-next-line jest/no-try-expect
+				expect(error.message).toInclude("Property '.tokenID' must NOT have more than 16 character");
+			}
+		});
+	});
+
+	describe('initGenesisState', () => {
+		let blockHeader: BlockHeader;
+		let blockExecuteContext: GenesisBlockExecuteContext;
+		let stateStore: PrefixedStateReadWriter;
+
+		beforeEach(() => {
+			stateStore = new PrefixedStateReadWriter(new InMemoryPrefixedStateDB());
+			blockHeader = createBlockHeaderWithDefaults({ timestamp: 1234 });
+			blockExecuteContext = createGenesisBlockContext({
+				header: blockHeader,
+				stateStore,
+			}).createInitGenesisStateContext();
+		});
+
+		it('should store genesis header timestamp', async () => {
+			await rewardModule.initGenesisState(blockExecuteContext);
+			const { timestamp } = await rewardModule.stores
+				.get(EndOfRoundTimestampStore)
+				.get(blockExecuteContext, EMPTY_BYTES);
+
+			expect(timestamp).toEqual(1234);
+		});
+	});
+
+	describe('beforeTransactionsExecute', () => {
+		let blockExecuteContext: BlockExecuteContext;
+		let generatorAddress: Buffer;
+		let standbyValidatorAddress: Buffer;
+		let stateStore: PrefixedStateReadWriter;
+
+		const activeDelegate = 4;
+		const minimumReward =
+			(BigInt(defaultConfig.brackets[0]) *
+				BigInt(defaultConfig.factorMinimumRewardActiveDelegates)) /
+			DECIMAL_PERCENT_FACTOR;
+		const totalRewardActiveDelegate = BigInt(defaultConfig.brackets[0]) * BigInt(activeDelegate);
+		const ratioReward = totalRewardActiveDelegate - minimumReward * BigInt(activeDelegate);
+
+		beforeEach(async () => {
+			generatorAddress = utils.getRandomBytes(20);
+			standbyValidatorAddress = utils.getRandomBytes(20);
+			stateStore = new PrefixedStateReadWriter(new InMemoryPrefixedStateDB());
+			const blockHeader = createBlockHeaderWithDefaults({
+				height: defaultConfig.offset,
+				generatorAddress,
+			});
+			blockExecuteContext = createBlockContext({
+				stateStore,
+				header: blockHeader,
+			}).getBlockExecuteContext();
+			await rewardModule.stores
+				.get(EndOfRoundTimestampStore)
+				.set(blockExecuteContext, EMPTY_BYTES, { timestamp: blockHeader.timestamp - 100000 });
+			const validators = [
+				{ address: generatorAddress, bftWeight: BigInt(20) },
+				{ address: utils.getRandomBytes(20), bftWeight: BigInt(30) },
+				{ address: utils.getRandomBytes(20), bftWeight: BigInt(40) },
+				{ address: utils.getRandomBytes(20), bftWeight: BigInt(10) },
+				{ address: standbyValidatorAddress, bftWeight: BigInt(0) },
+			];
+
+			(validatorsMethod.getValidatorsParams as jest.Mock).mockResolvedValue({ validators });
+			(dposMethod.getNumberOfActiveDelegates as jest.Mock).mockReturnValue(activeDelegate);
+		});
+
+		it('should store minimal reward for active delegates when full round is forged', async () => {
+			// Round is already completed once
+			const generatorMap = new Array(103).fill(0).reduce(prev => {
```

Comment:
Define a constant such as `ROUND_LENGTH` instead of using a literal integer 103. Both here and when initializing `dposMethod`. It makes it clearer what 103 means, and it's easier to maintain if we ever want to test with different round lengths.